netlify/cli
# ep/add deploy int command

- Add @netlify/sdk as a dependency.
- Import init method from Netlify SDK and invoke it
- Add await
- More lint things
- Add 'int' as alias for 'integration'
- Fix import
- Add build command
- Add the dev and preview commands
- Rename
- Conditionally add the integration command based on a feature flag.
- Rename files
- Forgot to uncomment this logic
- Add skeleton of deploy command for netlify integrations
- WIP commit. General outline of the integration deploy command
- WIP commit
- Pin the version of configstore to 5.
- Get integration information from integration service when site is linked
- Update slug automatically if it differs from jigsaw since it's the source of truth.
- update packages
- WIP commit
- Add logic for updating the integration info on remote.
- Add error handling on request to remote
- Add logic for registering an integration on the remote
- Add the deploying of the integration site
- WIP commit
- Slight reorg of code. Still needs further refactoring
- Update package.json
- Move register functionality to it's own function
- Move update functionality to it's own function
- Support env variables
- Make some changes based on initial code walkthrough with team
- Remove outdated comment
- Fix lint issues
- Trying to debug error in github actions
- Revert change made for debugging
- Update reference to @netlify/sdk
- Show error message from remote when one is returned
- add exit statements that were missing
- chore: run docs build
- chore: lintfix
- chore: prettier
- feat: convert to command:subcommand as is in the rest of the cli
- docs: add subcommands in docs
- chore: rename folder to match command name
- chore: prettier
- test: test for integration:init
- chore: remove unused import
- test: dev command
- test: add tests to make sure sdk is being called
- chore: update version of sdk
- chore: remove mystery markdown-magic
- test: starting on tests for deploy
- test: trying to get these mocks going
- test: more progress on test
- feat: add integrationLevel to integration deploy
- Exit process if required fields are missing
- Move reading the integration configuration to earlier in the order of operations
- Add integrationLevel to what's written in the yaml file
- test: passing test of deployment
- chore: prettier
- Add better error messaging.
- Run prettier
- Update link
- Support passing in the build flag to the int:deploy command
- chore: update sdk so integrationlevel validation works correctly
- Changes from docs review
- lint
- Missed including the level
- docs: docs review rewording
- feat: remove integration:init command
- docs: doc review changes
- docs: run docs build
- chore: format
- docs: remove beta
- feat: remove all but integration:deploy
- docs: remove docs for removed commands
- chore: package lock merge conflicts
- chore: merge conflict
- chore: update sdk version
- feat: add logging for finished, and be clearer in the deploy function
- feat: add shortened url
- fix: missing imports
